ape_safe/_cli.py

```diff
@@ -1,135 +1,224 @@
-def remove(cli_ctx, alias):
-    safe_container = accounts.containers["safe"]
-
-    if alias not in safe_container.aliases:
-        raise
+def remove(cli_ctx: SafeCliContext, alias):
+    if alias not in cli_ctx.safes.aliases:
+        raise BadArgumentUsage(f"There is no safe with the alias `{alias}`.")
 
-    address = safe_container.load_account(alias).address
+    address = cli_ctx.safes.load_account(alias).address
     if click.confirm(f"Remove safe {address} ({alias})"):
-        safe_container.delete_account(alias)
+        cli_ctx.safes.delete_account(alias)
 
+    cli_ctx.logger.success(f"Safe '{address}' ({alias}) removed.")
 
-@cli.command(
-    cls=NetworkBoundCommand, short_help="See pending transactions for a locally-tracked Safe"
-)
+
+# NOTE: The handling of the `--execute` flag in the `pending` CLI
+#    all happens here EXCEPT if a pending tx is executable and no
+#    value of `--execute` was provided.
+def _handle_execute_cli_arg(ctx, param, val):
+    # Account alias - execute using this account.
+    if val is None:
+        # Was not given any value.
+        # If it is determined in `pending` that a tx can execute,
+        # the user will get prompted.
+        # Avoid this by always doing `--execute false`.
+        return val
+
+    elif val in ctx.obj.account_manager.aliases:
+        return ctx.obj.account_manager.load(val)
+
+    # Account address - execute using this account.
+    elif val in ctx.obj.account_manager:
+        return ctx.obj.account_manager[val]
+
+    # Saying "yes, execute". Use first "local signer".
+    elif val.lower() in ("true", "t", "1"):
+        return True
+
+    # Saying "no, do not execute", even if we could.
+    elif val.lower() in ("false", "f", "0"):
+        return False
+
+    raise BadOptionUsage(
+        "--execute", f"`--execute` value '{val}` not a boolean or account identifier."
+    )
+
+
+@cli.command(cls=NetworkBoundCommand)
+@safe_cli_ctx
 @network_option()
+@click.option("sign_with_local_signers", "--sign", is_flag=True)
+@click.option("--execute", callback=_handle_execute_cli_arg)
 @existing_alias_argument(account_type=SafeAccount)
-def pending(network, alias):
-    safe = accounts.load(alias)
-    local_signers = set(signer for signer in accounts if signer.address in safe.signers)
-    if local_signers:
-        click.echo("Local Signer(s) detected!")
-        sign_with_local_signers = click.confirm("Do you want to sign unconfirmed transactions")
+def pending(cli_ctx: SafeCliContext, network, sign_with_local_signers, execute, alias) -> None:
+    """
+    View pending transactions for a Safe
+    """
+
+    _ = network  # Needed for NetworkBoundCommand
+    safe = cli_ctx.account_manager.load(alias)
+    submitter: Optional[AccountAPI] = None
+
+    if execute is True:
+        if not safe.local_signers:
+            cli_ctx.abort("Cannot use `--execute TRUE` without a local signer.")
+
+        submitter = get_user_selected_account(account_list=safe.local_signers)
```

Comment:
wonder if it's better to encode all of this logic in a callback that returns a `Optional[AccountAPI]` value to use in the script

Reply:
I donno if you can put this part in the callback because it requires the safe, which comes from a different option.

Reply:
OK - got this in the callback.